services/wireguard/service.go

```diff
@@ -111,41 +132,82 @@ func (manager *Manager) Stop() error {
 		return nil
 	}
 
+	if err := manager.wgClient.Close(); err != nil {
+		return err
+	}
+
+	if err := manager.deleteInterface(interfaceName); err != nil {
+		return err
+	}
+
 	manager.process.Done()
 	manager.isStarted = false
 	log.Info(logPrefix, "Wireguard service stopped")
 	return nil
 }
 
-func setupWireguard(publicIP string) (Config, error) {
-	// TODO initialize wireguard interface.
-	// out, err := exec.Command("ip", "link", "add", "dev", "wg0", "type", "wireguard").CombinedOutput()
-	// out, err := exec.Command("ip", "address", "add", "dev", "wg0", "192.168.100.1/24").CombinedOutput()
-
-	// TODO configure wireguard interface.
-	// client, err := wireguardctrl.New()
-	// if err != nil {
-	// 	return Config{}, err
-	// }
-
-	// TODO wireguard device configuration like private key, listen port, peer list should survive through restarts.
-	// TODO we need to have some persistent storage for it.
-	// client.ConfigureDevice("wg0", wgtypes.Config{
-	// 	PrivateKey: "8C6Pp0cI2tgLeYOMVbnMMLl/zN2blFK+OWGaIxX0bHY=",
-	// 	ListenPort: 52820,
-	// 	Peers:      nil,
-	// })
-
-	// TODO if the wireguard interface already configured we can get required parameters from it.
-	// device, err := client.Device("wg0")
-	// if err != nil {
-	// 	return Config{}, err
-	// }
+func (manager *Manager) initInterface(name string) error {
```

Comment:
Real tunnel device creation is really OS specific and must be abstracted away from ServiceManager - on linux we can use this approach, on other OSes - userspace impl.

Reply:
It will possibly break e2e

Reply:
Agree, I'll add abstraction level. And for E2E tests we will need to test every transport separately. We have a task for the E2E test for wireguard somewhere.